shared_model/interfaces/iroha_internal/transaction_sequence.cpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/iroha_internal/transaction_sequence.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    bool TransactionComparator::operator()(
+        const std::shared_ptr<Transaction> &lhs,
+        const std::shared_ptr<Transaction> &rhs) const {
+      // implement fair logic here
```

Comment:
Implementation of this method will be not obvious because the operator must stay in butches in one order. I can't imagine such implementation. Do you have an algorithm for that?

Reply:
1. Every transaction without batch should be less than any transaction without batch.
2. Between two transactions from different batches, the transaction from the batch where tx_hashes is less than tx_hashes from another batch will be less
3. Two transactions from the same batch are sorted using tx_hashes array